src/lang.rs

```diff
@@ -352,6 +352,28 @@ named!(sourced_expr<Span, (String, Expr)>, ws!(
         )
 )));
 
+
+
+
+named!(operator_name<Span, &str>,
+    do_parse!(
+        return_error!(SyntaxErrors::DelimiterStart.into(),
+            alt!(ws!(alt!(tag!("parse") | tag!("where") | tag!("json") | tag!("fields") | tag!("where") | tag!("limit") | tag!("total"))) |
+            // If we exhaust all other possbilities, consume a word and return an error. We won't
+            // find `tag!("a")` after an identifier, so that's a guaranteed to fail and produce `not an operator`
+            preceded!(take_while!(is_ident), return_error!(SyntaxErrors::NotAnOperator.into(), tag!("a"))))) >>
```

Comment:
this is a hack to get the error position into the right place -- we need to actually consume some input to move it over.